tools/code_format/check_format.py

```diff
@@ -114,7 +114,8 @@
 # Header files that can throw exceptions. These should be limited; the only
 # valid situation identified so far is template functions used for config
 # processing.
-EXCEPTION_ALLOWLIST = ("./source/common/config/utility.h")
+EXCEPTION_ALLOWLIST = ("./source/common/config/utility.h",
+                       "./source/extensions/filters/http/common/factory_base.h")
```

Comment:
@ahedberg @htuch 

We may have some problems here depending on the intent behind this check.  It seems to me that headers in this list may need to be restricted so they are not included from other headers in other to prevent multiple definitions in builds that partially disable exceptions at a per-library basis. That said, there seems to be no restrictions in place that control where source/common/config/utility.h is included from or the use of those functions in header files.

This PR introduces an exception that is thrown from a base class method of a class that seems to be widely used.  I think this change could break compilation of libraries that disable exceptions.

Reply:
Another option is we can fix it here https://github.com/envoyproxy/envoy/blob/b23ee3b438050228b7c7b0097d182ac828fa2247/source/common/router/config_impl.cc#L1492-L1502

Reply:
If we decide to have `createRouteSpecificFilterConfigTyped` throw an exception, the `throw` statement should be in the .cc file instead of the .h file. Then we don't need to add it to the exception allowlist.

Reply:
FactoryBase is a template class, the throw can't be moved to a cc file (well, except possibly by adding a helper function)

Returning null and doing the validation in config_impl.cc seems attractive.  Is that the only place where this factory method is called from?

Reply:
> Returning null and doing the validation in config_impl.cc seems attractive. Is that the only place where this factory method is called from?

Yes, that is the only place I found at least.

Reply:
I think `config_impl.cc` is more robust in any case, since regardless of inheritance hierarchy, if there is no config, we can fire the exception.

Reply:
It would be really helpful to some of us if exceptions were only thrown from cc files.  Could you do the new config validation in config_impl.cc instead of the template factory class?

Reply:
Yes, I will update, thanks for the feedback!